### DISCUSSION TITLE
"Extract function" action does not extract comments (go)

### DISCUSSION THREAD

**Discussion by ygabuev** [helix-editor/helix#9934] [General]
### Summary

When running an "extract function" code action on a chunk of Golang code, the comments are not moved to the extracted function but stay in their original place. I am not sure if this is the helix bug or `gopls` bug. I don't have any extra go-specific configuration in `languages.toml`.

### Reproduction Steps

Start with this code:
```go
func main() {
	num := 1
	// set foo to foo
	foo := "foo"
	// set bar to bar
	bar := "bar"
}
```
Select all of the contents of `main()`, press `<space>a`, choose "Extract function". I get the following as the result:
```go
func main() {
	// set foo to foo
	// set bar to bar
	newFunction()
}

func newFunction() {
	num := 1

	foo := "foo"

	bar := "bar"
}
```
Notice that the comments `// set ...` are not moved to the extracted function.

### Platform

Arch Linux, x86-64

### Terminal Emulator

alacritty 0.13.1

### Installation Method

Arch Linux repositories

### Helix Version

23.10

**Comment by pascalkuthe** [comment]
We don't control code actions. They are handled by the language server. We just display and execute whatever the LSP sends. You need to report this to the go repo.
